sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -211,49 +234,59 @@ trait CostModel {
    * @param plan Operator
    * @return (cpuCost, gpuCost)
    */
-  def applyCost(plan: SparkPlanMeta[_]): (Double, Double)
+  def getCost(plan: SparkPlanMeta[_]): Double
 
-  /**
-   * Determine the cost of transitioning data from CPU to GPU for a specific operator
-   * @param plan Operator
-   * @return Cost
-   */
-  def transitionToGpuCost(plan: SparkPlanMeta[_]): Double
-
-  /**
-   * Determine the cost of transitioning data from GPU to CPU for a specific operator
-   */
-  def transitionToCpuCost(plan: SparkPlanMeta[_]): Double
 }
 
-class DefaultCostModel(conf: RapidsConf) extends CostModel {
+class CpuCostModel(conf: RapidsConf) extends CostModel {
 
-  def transitionToGpuCost(plan: SparkPlanMeta[_]) = {
-    // this is a placeholder for now - we would want to try and calculate the transition cost
-    // based on the data types and size (if known)
-    conf.defaultTransitionToGpuCost
+  def getCost(plan: SparkPlanMeta[_]): Double = {
+    val rowCount = RowCountPlanVisitor.visit(plan).map(_.toDouble)
+      .getOrElse(conf.defaultRowCount.toDouble)
+    val costPerRow = plan.conf.getOperatorCost(plan.wrapped.getClass.getSimpleName).getOrElse {
+      plan.wrapped match {
+        case _: ProjectExec =>
+          // the cost of a projection is the sum of its expressions
+          plan.childExprs
+            .map(expr => exprCost(expr.asInstanceOf[BaseExprMeta[Expression]]))
+            .sum
+        case _ => 1.0
+      }
+    }
+    rowCount * costPerRow
   }
 
-  def transitionToCpuCost(plan: SparkPlanMeta[_]) = {
-    // this is a placeholder for now - we would want to try and calculate the transition cost
-    // based on the data types and size (if known)
-    conf.defaultTransitionToCpuCost
+  private def exprCost[INPUT <: Expression](expr: BaseExprMeta[INPUT]): Double = {
+    // always check for user overrides first
+    expr.conf.getExpressionCost(expr.getClass.getSimpleName).getOrElse {
+      expr match {
+        case _ =>
+          // many of our BaseExprMeta implementations are anonymous classes so we look directly at
+          // the wrapped expressions in some cases
+          expr.wrapped match {
+            case _: AttributeReference => 0.01
```

Comment:
nit: instead of a magic number of 0.01 it would be nice to have a common value somewhere. for these